infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/SinkWriter.java

```diff
@@ -28,22 +28,23 @@
 @Slf4j
 public abstract class SinkWriter {
 
-    // Configuration for LR in Source / Sink cluster.
-    private final LogReplicationConfigManager configManager;
-
     private final ISerializer protobufSerializer;
 
     private final ReplicationSession session;
 
+    // Replication context that provides configuration for LR in Source / Sink cluster.
+    LogReplicationContext replicationContext;
+
 
     // Limit the initialization of this class only to its children classes.
-    SinkWriter(LogReplicationConfigManager configManager, ReplicationSession session) {
-        this.configManager = configManager;
+    SinkWriter(LogReplicationContext replicationContext, ReplicationSession session) {
+        this.replicationContext = replicationContext;
         this.session = session;
 
         // The CorfuRuntime in LogReplicationConfigManager used to get the config fields from registry
         // table, and the protobufSerializer is guaranteed to be registered before initializing SinkWriter.
-        this.protobufSerializer = configManager.getRuntime().getSerializers().getSerializer(PROTOBUF_SERIALIZER_CODE);
+        this.protobufSerializer = replicationContext.getConfigManager().getRuntime().getSerializers()
```

Comment:
is it better to encapsulate all this in context in API getSerializers()

Reply:
New interface and java doc added. Thanks!